### DISCUSSION TITLE
Trying to using exception lifecycle

### DISCUSSION THREAD

**Discussion by killallskywalker** [livewire/livewire#8373] [1. Help]
Currently just found out that in livewire have exception lifecycle hook . I try to use it , somehow it did't go through the exception function . 

```

  in my component , on mount i trigger the ModelNotFoundException to see if the hook below  are triggered or not 

   public function exception($e, $stopPropagation) {
        if($e instanceof ModelNotFoundException) {
            dd('Not  execute');
        }
    }
```

Is there anything that i miss ? As per in doc it state - 

` The exception() hook allows you to do just that: you can perform check on the $error, and use the $stopPropagation parameter to catch the issue. `

Thanks !

**Comment by lrljoe** [comment]
Make sure that you've got that exception available in your file

and call

```
$stopPropagation();
```

**Comment by marcusmoore** [comment]
I was able to throw a `ModelNotFoundException` from `mount` and catch it in the `exception` method but I did notice that it doesn't seem to work for exceptions thrown in computed properties.

**Reply by lrljoe** [reply]
Can you share a simple example component demonstrating this, as it seems to be working for me in a basic example.

**Reply by marcusmoore** [reply]
Here is a [Wirebox](https://wirebox.app/b/x9o3k) with a simple example. Throwing an exception from an action (triggered by clicking the button) results in landing in the `exception` method but throwing it from the computed property (you'll have to uncomment line 24) doesn't.

Here it is in [gist](https://gist.github.com/marcusmoore/a5d9b0962eaf291b9e43597f95452559) form in case it's easier to reference.

**Reply by lrljoe** [reply]
Hmm

Catching it and passing it to the exception **does** work, so I suspect it's down to how Computed Properties are actually dealt with (via invading).

```
    #[Computed]
    public function somethingComputed()
    {
        // 2: uncomment the "throw new Exception" line and save the component (cmd+s).
        // Wirebox will reload the preview and this exception will render 
        // without landing in the exception() method.
        try {
            throw new Exception('from a computed property');    
        }
        catch (Exception $e)
        {
            $this->exception($e, true);
        }
        
        
        return 'I am text from a computed property';
    }
```

**Reply by marcusmoore** [reply]
Ah good thinking @lrljoe! 😄

**Reply by lrljoe** [reply]
Not a proper solution, but at least gives you a break-fix for now, will take a look at the exception capture for Computed attributes when I get a chance (unless someone else beats me to it), but hopefully you can at least progress for now!

For what it's worth, you could extend the Computed attribute, so that it exhibits that behaviour by default.

**Reply by lrljoe** [reply]
@marcusmoore  - Just to note that this PR should help on the way to addressing this, as it starts to implement lifecycle hooks into the Form #8682

**Reply by marcusmoore** [reply]
@lrljoe Good thinking on extending the attribute. I might give that a try when I circle back to the work that this popped up in. I'll keep an eye on that PR as well.

Thanks for following up 😄

**Reply by faraweilyas** [reply]
dang! there's still no work around for this in computed properties

**Reply by ghabriel25** [reply]
@marcusmoore @faraweilyas fixed in https://github.com/livewire/livewire/pull/10543
